asar is a PiTA to install in Windows (7 64 bit here)

A dependency on Python and an ancient version of the Net Frameworks makes this a royal pain-in-the-arse to install on Windows (I've yet to manage to find the exact Framework it's asking for)

Tips apprec.

What I get (it's not easy to copy this from terminal) is....

C:\Users\john>npm install asar
\
> native-mate@0.2.6 install C:\Users\john\node_modules\asar\node_modules\chromiu
m-pickle\node_modules\native-mate
> echo

ECHO is on.

> chromium-pickle@0.1.4 install C:\Users\john\node_modules\asar\node_modules\chr
omium-pickle
> node-gyp rebuild


C:\Users\john\node_modules\asar\node_modules\chromium-pickle>node "E:\nodejs\nod
e_modules\npm\bin\node-gyp-bin\\..\..\node_modules\node-gyp\bin\node-gyp.js" reb
uild
Warning: Missing input files:
C:\Users\john\node_modules\asar\node_modules\chromium-pickle\node_modules\native
-mate\tests\compilation.cc
Building the projects in this solution one at a time. To enable parallel build,
please add the "/m" switch.
MSBUILD : error MSB3428: Could not load the Visual C++ component "VCBuild.exe".
 To fix this, 1) install the .NET Framework 2.0 SDK, 2) install Microsoft Visua
l Studio 2005 or 3) add the location of the component to the system path if it
is installed elsewhere.  [C:\Users\john\node_modules\asar\node_modules\chromium
-pickle\build\binding.sln]
MSBUILD : error MSB3428: Could not load the Visual C++ component "VCBuild.exe".
 To fix this, 1) install the .NET Framework 2.0 SDK, 2) install Microsoft Visua
l Studio 2005 or 3) add the location of the component to the system path if it
is installed elsewhere.  [C:\Users\john\node_modules\asar\node_modules\chromium
-pickle\build\binding.sln]
gyp ERR! build error
gyp ERR! stack Error: `C:\Windows\Microsoft.NET\Framework\v4.0.30319\msbuild.exe
` failed with exit code: 1
gyp ERR! stack     at ChildProcess.onExit (E:\nodejs\node_modules\npm\node_modul
es\node-gyp\lib\build.js:267:23)
gyp ERR! stack     at ChildProcess.emit (events.js:98:17)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (child_process.js:820:
12)
gyp ERR! System Windows_NT 6.1.7601
gyp ERR! command "node" "E:\\nodejs\\node_modules\\npm\\node_modules\\node-gyp\\
bin\\node-gyp.js" "rebuild"
gyp ERR! cwd C:\Users\john\node_modules\asar\node_modules\chromium-pickle
gyp ERR! node -v v0.10.35
gyp ERR! node-gyp -v v1.0.1
gyp ERR! not ok

npm ERR! chromium-pickle@0.1.4 install: `node-gyp rebuild`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the chromium-pickle@0.1.4 install script.
npm ERR! This is most likely a problem with the chromium-pickle package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node-gyp rebuild
npm ERR! You can get their info via:
npm ERR!     npm owner ls chromium-pickle
npm ERR! There is likely additional logging output above.
npm ERR! System Windows_NT 6.1.7601
npm ERR! command "E:\\nodejs\\\\node.exe" "E:\\nodejs\\node_modules\\npm\\bin\\n
pm-cli.js" "install" "asar"
npm ERR! cwd C:\Users\john
npm ERR! node -v v0.10.35
npm ERR! npm -v 1.4.28
npm ERR! code ELIFECYCLE
npm ERR! not ok code 0


On 24 February 2015 at 01:45, Paul Betts <notifications@github.com> wrote:
ASAR isn't written in C# at all - why are you getting prompted to install the .NET Framework?

—
Reply to this email directly or view it on GitHub.

--- a/.travis.yml
+++ b/.travis.yml
@@ -1,4 +1,4 @@
 language: node_js
 node_js:
-  - "0.11"
+  - "0.12"
   - "0.10"

--- a/lib/disk.js
+++ b/lib/disk.js
@@ -2,7 +2,7 @@ var fs = require('fs');
 var path = require('path');
 
 var mkdirp = require('mkdirp');
-var pickle = require('chromium-pickle');
+var pickle = require('chromium-pickle-js');
 
 var Filesystem = require('./filesystem');
 

--- a/package.json
+++ b/package.json
@@ -26,7 +26,7 @@
     "clean": "rm -rf tmp/"
   },
   "dependencies": {
-    "chromium-pickle": "0.1.4",
+    "chromium-pickle-js": "0.1.0",
     "commander": "2.3.0",
     "cuint": "0.1.5",
     "minimatch": "2.0.4",

